symbulate/tests/test_distributions.py

```diff
@@ -10,25 +10,683 @@
+        self.assertTrue(pval > .01)
+
+class TestPareto(unittest.TestCase):
+
+    def test_Pareto_check_mean(self):
+        x = stats.pareto(b=-3) 
+        math.isnan(x.mean())
+
+    def test_Pareto_to_Exponential(self):
+        X = RV(Pareto(b=1))
+        sims = (log(X + 1)).sim(Nsim)
+        cdf = stats.expon(scale=1).cdf
+        pval = stats.kstest(sims, cdf).pvalue
+        self.assertTrue(pval > .01)
+
+
+class TestWeibull(unittest.TestCase):
+    def test_Weibull_to_Exponential(self):
+        '''X = RV(Weibull(scale=1, c=10))
+           sims = X.sim(Nsim)
+           cdf = stats.expon(scale=1/10).cdf
+           pval = stats.kstest(sims, cdf).pvalue
+           self.assertTrue(pval > .01)
+        '''
+        pass
+
+class TestRayleigh(unittest.TestCase):
+
+    def test_Rayleigh_Normal(self):
+        A, B = RV(Normal(mean=0, var=1) * Normal(mean=0, var=1))
+        sims = (A ** 2 + B ** 2).apply(sqrt).sim(Nsim)
+        cdf = stats.rayleigh.cdf
+        pval = stats.kstest(sims, cdf).pvalue
+        self.assertTrue(pval > .01)
+
+    def test_Rayleigh_to_Chi(self):
+        X = RV(Rayleigh())
+        sims = X.sim(Nsim)
+        cdf = stats.chi(df=2).cdf
+        pval = stats.kstest(sims, cdf).pvalue
+        self.assertTrue(pval > .01)
+
+
+class TestMultivariateNormal(unittest.TestCase):
+     
+    def test_MultivariateNormal_mean_cov_error(self):
+        self.assertRaises(Exception, 
+             lambda: MultivariateNormal(mean=[2], cov=[[2,2], [3,4]]))
+
+    def test_MultivariateNormal_cov_square_error(self):
+        self.assertRaises(Exception, 
+             lambda: MultivariateNormal(mean=[2,4], cov=[[2,4,5],[2,1]]))
+
+
+class TestBivariateNormal(unittest.TestCase):
+    
+    def test_BivariateNormal_error1(self):
+        self.assertRaises(Exception, lambda: 
+             BivariateNormal(mean1=3, mean2=4, sd1=-3, sd2=3, corr=0.9))
+
+    def test_BivariateNormal_error2(self):
+        self.assertRaises(Exception, lambda: 
+             BivariateNormal(mean1=3, mean2=4, sd1=3, sd2=3, corr=1.1))
+ 
+    def test_LinCom_BivNormal(self):
+        X, Y = RV(BivariateNormal(mean1=30, mean2=50, sd1=8, sd2=6, corr=-0.4))
+        Z = 4 * X - 2 * Y
+        Z_mean = 4 * 30 + (-2) * 50
+        Z_var = 16 * 64 + 4 * 36 + 2 * (4) * (-2) * (-0.4 * 8 * 6)
+        sims = Z.sim(Nsim)
+        cdf = stats.norm(loc=Z_mean, scale=sqrt(Z_var)).cdf
+        pval = stats.kstest(sims, cdf).pvalue
+        self.assertTrue(pval > .01)
+    
+    def test_BivNormal_condDistr_r(self):
+        results = []
+        corr_list = list(np.arange(-0.9, 0.9, 0.1))
+        for c in corr_list:
+            X,Y = RV(BivariateNormal(mean1=20, mean2=10, sd1=3, sd2=5, corr=c))
+            sims = (Y | (abs(X - 21) < 0.1)).sim(500)
+            cdf = stats.norm(loc=10 + c * 5 / 3 * (21 - 20), 
+                             scale=5 * sqrt(1 - c ** 2)).cdf
+            pval = stats.kstest(sims, cdf).pvalue
+            results.append((pval > 0.01))
```

Comment:
You don't need to create the results list. Just call `self.assertTrue(pval > 0.01)` inside the for loop....